### PR TITLE
gnrc_netreg: enhance doc on entry initialization

### PR DESCRIPTION
**Pull request by miri64**
The `gnrc_netreg` documentation was a little bit outdated. This fixes that.

### CODE REVIEW

File: sys/include/net/gnrc/netreg.h [hunk 1]
```diff
@@ -212,6 +218,11 @@ typedef struct gnrc_netreg_entry {
 void gnrc_netreg_init(void);
 
 /**
+ * @name    Dynamic entry initialization functions
```
Comment (smlng) [inline sys/include/net/gnrc/netreg.h]: inconsistent text: for static (above) it says `macros` not `functions`

File: sys/include/net/gnrc/netreg.h [hunk 2]
```diff
@@ -284,8 +296,11 @@ static inline void gnrc_netreg_entry_init_cb(gnrc_netreg_entry_t *entry,
  *
  * @param[in] type      Type of the protocol. Must not be < GNRC_NETTYPE_UNDEF or
  *                      >= GNRC_NETTYPE_NUMOF.
- * @param[in] entry     An entry you want to add to the registry with
- *                      gnrc_netreg_entry_t::pid and gnrc_netreg_entry_t::demux_ctx set.
+ * @param[in] entry     An entry you want to add to the registry. This needs to
```
Comment (smlng) [inline sys/include/net/gnrc/netreg.h]: I find this addressing the reader using `you` a bit uncommon for documentation - I know it was there before, but still this about improving docu, right 😉 Hence, propose to change to something like: `New registry entry to be added.`

File: sys/include/net/gnrc/netreg.h [hunk 3]
```diff
@@ -80,6 +80,11 @@ typedef enum {
 #define GNRC_NETREG_DEMUX_CTX_ALL   (0xffff0000)
 
 /**
+ * @name    Static entry initialization macros
+ * @anchor  net_gnrc_netreg_init_static
```
Comment (smlng) [inline sys/include/net/gnrc/netreg.h]: side comment: nice tag, we should use this more often in the RIOT docu

File: sys/include/net/gnrc/netreg.h (hunk 1 above)
Comment (miri64) [inline sys/include/net/gnrc/netreg.h]: Because they are macros... This

```C
void some_name(int param);
```

is a function definition. This

```C
#define some_name(param)    (param + 1)
```

is a macro declaration ;-)

Comment (smlng) [inline sys/include/net/gnrc/netreg.h]: oops, didn't expand the code - just thought the lines should match.

File: sys/include/net/gnrc/netreg.h (hunk 2 above)
Comment (miri64) [inline sys/include/net/gnrc/netreg.h]: This would require the rest of the doc in this file to be changed as well. This would lead to unrelated changes => this is unrelated and should be dealt with in a separate PR, if you want it ;-).

Comment (smlng) [inline sys/include/net/gnrc/netreg.h]: okay fine, nevermind - its also my personal preference in docu style
